### ISSUE TITLE
[Experiment] Add support for wide screen mode

### ISSUE THREAD

**Issue by JNechaevsky** [Russian-Doom/russian-doom#90]
This is absolutely lovely feature, with both good and awful sides. 

**Good:**
- The game will be playing much and far better on wide screen monitors.
- People will be happy about it.
- I **want** it.

**Bad:**
- Will require tons of `SCREENWIRDTH` / `SCREENWIDTH_WIDE` and probably `ORIGWIDTH` / `ORIGWIDTH_WIDE` conditions.
- Will require additional conditions for some graphical elements for horizontal offset correction.
- Not sure how to fix bordered view.
- Something somewhere somehow is dependent in Doom's monster's behaviour. With modified `SCREENWIRDTH` I've a got a demo desync in Doom 1.

**Verdict:**
- It should be available in Display Options in Setup executable, as well as variable in config file, and must be declared strictly as experimental. Visually it should be polished, but in terms of code... Who cares.

In general, it's a gross hack, since wide mode will also be hardcoded, just like original screen mode. But it's benefit is huge. Hopefully, it should work fine in 16:9 and 16:10 monitors.

**Magical numbers:**
```
#define ORIGWIDTH_WIDE      426
#define SCREENWIDTH_WIDE    (ORIGWIDTH_WIDE << hires)
#define WIDE_OFFSET         53 // (426-320)/2 = 53
```

**Comment by JNechaevsky** [comment]
After ~eight hours of `(widescreen ? SCREENWIDTH_WIDE : SCREENWIDTH)` carnival now it's obvious that this not way to go. The code is turning into trash, some things getting working, some getting broken, and it's like going around in circle. So screw it.

But that's not the reason to give up. Better I make a custom branch / executable, this must be an easiest and safest approach. And still, it should be declared as experimental.

**Comment by JNechaevsky** [comment]
> Something somewhere somehow is dependent in Doom's monster's behaviour. With modified SCREENWIRDTH I've a got a demo desync in Doom 1.

Figured out, It's here. This `SCREENWIDTH` must not be changed.

P_AimLineAttack @@ p_map.c
```
    // can't shoot outside view angles
    topslope = (SCREENHEIGHT/2)*FRACUNIT/(SCREENWIDTH/2);	
    bottomslope = -(SCREENHEIGHT/2)*FRACUNIT/(SCREENWIDTH/2);
```

**Comment by JNechaevsky** [comment]
Things left to do:

- Fix magic values of vertical hitscan aiming, so bullets will shoot into the crosshair. 
- Fix game crush on bunny scrolling screen. AFAIR, same problem was in very first versions of ZDoom, which is interesting.
- Reconstruct standard gray HUD to use in "fullscreen" mode. Mostly done, need some cleanup.
- Disallow to use/have bordered view, but keep `screenblocks` values same and compatible with original 4:3 executable.
- Check 16:10 monitor and HDMI TV. Maybe it's worth to disable fixing aspect ratio correction to perfectly fill whole screen area.
- Finally, be happy about it, but there are also Heretic and Hexen.

**Comment by JNechaevsky** [comment]
Well, it seems to be done, all elements and stuff are on right places. Available screen sizes are:

* [Screenblocks = 9](https://user-images.githubusercontent.com/21193394/47934322-158ad880-dee8-11e8-9695-6bfe0976a0a4.png). Traditional HUD + borders. Those borders are always drawn on automap. Values below are not accepted because of A) there is no proper bezel drawing, B) for compatibility with 4:3 executable sizes.
* [Screenblocks = 10](https://user-images.githubusercontent.com/21193394/47934403-3ce1a580-dee8-11e8-8100-979a909690be.png). Traditional HUD.
* [Screenblocks = 11](https://user-images.githubusercontent.com/21193394/47934447-54b92980-dee8-11e8-8b50-f24a230d34b4.png). Signed Crispy HUD™ with player's face.
* [Screenblocks = 12](https://user-images.githubusercontent.com/21193394/47934493-72868e80-dee8-11e8-82b1-7996f17b09b5.png). Signed Crispy HUD™ w/o player's face.
* [Screenblocks = 13](https://user-images.githubusercontent.com/21193394/47934536-87fbb880-dee8-11e8-9124-7ccb6bb09659.png). Traditional Crispy HUD™.
* [Screenblocks = 14](https://user-images.githubusercontent.com/21193394/47934560-9fd33c80-dee8-11e8-952f-6d01eafaee85.png). Clean screen.

As about Heretic and Hexen - I have no luck with proper RAW screen rendering, and auto map is screwed up. Not sure I'll handle it, but neither I have big interest for doing this ...

**Comment by JNechaevsky** [comment]
Goddammit. This explains tendency why standard HUD is not existing w/o borders in source ports with wide screen support. This visual bug is a bit rare, and good thing I can fix it graphically.

![doom01](https://user-images.githubusercontent.com/21193394/47942215-2c3e2900-df02-11e8-9663-262e61fcbddb.png)

**Comment by JNechaevsky** [comment]
Fixed. Looking quite unusual, but it's better than having a leak. Positioning is still vanilla.

![sht2e0](https://user-images.githubusercontent.com/21193394/47949898-7273a680-df5b-11e8-8d8b-05bb867c49e6.png)
![doom11](https://user-images.githubusercontent.com/21193394/47949930-cc746c00-df5b-11e8-9e2a-e0fe52e559ae.png)

**Comment by JNechaevsky** [comment]
Tested wide mode on following devices:

- PC with 27" monitor @ 1920x1080
- Three notebooks (15" and 13") @ 1366x768
- TV (some kind") @ 1600x900

Everything is okay, I have no vertical/horizontal borders on the screen. Yay!

**Comment by JNechaevsky** [comment]
Done here. Separate executable will be included in 3.1 release, still marker as experimental.

How it actually looks: https://youtu.be/CX-t-OVq3fg
